Mage.Sets/src/mage/cards/e/ExtraordinaryJourney.java

```diff
@@ -0,0 +1,205 @@
+        return new ExtraordinaryJourney(this);
+    }
+}
+
+enum ExtraordinaryJourneyAdjuster implements TargetAdjuster {
+    instance;
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        ability.getTargets().clear();
+        ability.addTarget(new TargetCreaturePermanent(0, ManacostVariableValue.ETB.calculate(game, ability, null)));
+    }
+}
+
+class ExtraordinaryJourneyEffect extends OneShotEffect {
+
+    ExtraordinaryJourneyEffect() {
+        super(Outcome.Neutral);
+        this.staticText = "exile up to X target creatures. For each of those cards, "
+                + "its owner may play it for as long as it remains exiled";
+    }
+
+    private ExtraordinaryJourneyEffect(final ExtraordinaryJourneyEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ExtraordinaryJourneyEffect copy() {
+        return new ExtraordinaryJourneyEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        List<Permanent> permanents = source
+                .getTargets().get(0).getTargets()
+                .stream()
+                .map(game::getPermanent)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        // exile up to X target creatures.
+        Effect effect = new ExileTargetEffect().setTargetPointer(new FixedTargets(permanents, game));
+        if(!effect.apply(game, source)) {
+            return false;
+        }
+
+        Set<Card> cards = permanents
+                .stream()
+                .map(p -> p.getMainCard())
+                .filter(Objects::nonNull)
+                .filter(card -> game.getState().getZone(card.getId()) == Zone.EXILED)
+                .collect(Collectors.toSet());
+
+        Set<Player> owners = cards
+                .stream()
+                .map(Card::getOwnerId)
+                .map(game::getPlayer)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+
+        // For each of those cards, its owner may play it for as long as it remains exiled.
+        for (Player owner : owners) {
+            String exileZoneName = "Exile — Can be played by " + owner.getName();
+            UUID exileZoneId = CardUtil.getExileZoneId(
+                    exileZoneName,
+                    game
+            );
+
+            ExileZone zone = game.getState().getExile().createZone(exileZoneId, exileZoneName);
+
+            for(Card card : cards) {
+                if (card.getOwnerId().equals(owner.getId())) {
+                    game.getExile().moveToAnotherZone(card, game, zone);
+                    CardUtil.makeCardPlayable(
+                            game, source, card, Duration.Custom,
+                            false, card.getOwnerId(), null
+                    );
+                }
+            }
+        }
+
+        return true;
+    }
+}
+
+class ExtraordinaryJourneyTriggeredAbility extends TriggeredAbilityImpl {
```

Comment:
You can probably simplify this ability, remember that the "one or more" wording doesn't actually matter since it's only once per turn so you can just make it a regular etb trigger